core/source/Dom.mint

```diff
@@ -590,4 +590,28 @@ module Dom {
   ) : Promise(Void) {
     `#{element}.scrollTo({ behavior: 'smooth', left: #{left}, top: #{top} })`
   }
+
+  /*
+  Returns the distance from the outer border of the element (including its
+  margin) to the left padding edge of the closest positioned ancestor element.
+
+    if let Just(div) = Document.getElementBySelector("div") {
+      Dom.offsetLeft(div)
+    }
+  */
+  fun offsetLeft (element : Dom.Element) : Number {
+    `#{element}.offsetLeft || -1`
```

Comment:
Returning `-1` is here is IMO totally unexpected and may lead to hard-to-debug bugs.
Also, AFAIU it's not coherent with this method documentation.

ditto below

Reply:
We have other functions that return values similarly https://github.com/mint-lang/mint/blob/master/core/source/Dom.mint#L371 https://github.com/mint-lang/mint/blob/dom-position-things/runtime/src/normalize_event.js#L93 We could return `Maybe(Number)` (and for all similar functions) but it would greatly increase the friction when using this.

I'll update the documentation comment to reflect this.

Reply:
If the element is not in the DOM, it returns `0`:
```
document.createElement("div").offsetLeft
```

So maybe returning `0` if it's not applicable is better?

Reply:
`0` sounds better to me, although it still seems more like a hack tbh...

Reply:
I guess it's good for now, yet it needs to be changed for other props as well.